### DISCUSSION TITLE
Debug VSCode netbox-plugin-tutorial

### DISCUSSION THREAD

**Discussion by via78** [netbox-community/netbox#11381] [Q&A]
Hi!  after i followed  the  tutorial steps  Plugin worked succesfully tryed to use visual studio code debug function and set  the launch.json ,but debug runnig  have Problem with find the  Postsqllib  . 
Is it because the NetBox files and plugin files are not in the same folder?

**Comment by dreng** [comment]
I set this up just last month in my environment and it works like a charm. It doesn't matter if you've got different folders for netbox and your plugin. That's the way it should be. I roughly followed this guide: https://code.visualstudio.com/docs/python/tutorial-django

Make sure you activated your venv and you ran "python setup.py develop", which links all the needed parts of your plugin into netbox and you set DEBUG = True in your configuration.py. This is described in the [netbox plugin tutorial](https://github.com/netbox-community/netbox-plugin-tutorial).

Following the linked guide above, in VS Code / Codium you have to 
1. Select an Interpreter (use "python" within your venv folder)
2. Run "Terminal: Create New Terminal"
3. Add a launch profile (aka launch.json)

No need to worry about postgresql libs or anything. Btw. I installed postgresql according to the [official netbox manual](https://docs.netbox.dev/en/stable/installation/1-postgresql/). No docker.
